### ISSUE TITLE
if/else conditions

### ISSUE THREAD

**Issue by luksak** [brettmc/docker-compose-generator#16]
It would be really useful to be able to have if/else conditions to add contents dynamically to `docker-compose.yml`

**Comment by brettmc** [comment]
Hi @luksak - thanks for the feedback.
I think I understand what you're after. I do something like this in a production deployment for a couple of projects, and I've found the simplest way to maintain deployments across multiple environments is to:
* break one big YAML into multiple smaller ones: a main file which is the main template and has everything that's common across all environments, then smaller per-environment files which contain only the bits have different YAML structure
* maintain per-environment .ini files for values that are fairly constant per-environment
* pass in really dynamic values (eg that might be generated from a commit hash, a CI/CD variable, etc) via the `-e` switch

I usually write a bash script wrapper which receives a couple of inputs from CI/CD to work out which YAML and .ini files to use and which variables to pass in.

Does that sound like it might suit your needs? Now that I've been using this tool in production for a couple of years, I'm overdue to improve the documentation and remove some cruft.

**Comment by luksak** [comment]
Thank you for your detailed reply!

I actually ended up doing the same thing; creating a bash script that collects snippets of services from several `.yml` files. That works great and is a much simpler solution than any existing project i could find.

I tried using this project as well, but my `.yml` fails validation, since the different parts depend on each other.

So in the end, just appending snippets using bash is a lot simpler and works great so far. As an alternative I'd rather go for a fully fledged templating language such as Twig to get more flexibility.
